src/chain/chain-generic-numerator.cc

```diff
@@ -98,246 +85,294 @@ GenericNumeratorComputation::GenericNumeratorComputation(
       }
 
       for (fst::ArcIterator<fst::StdVectorFst> aiter(
-               supervision_.e2e_fsts[seq], s);
+             supervision_.e2e_fsts[seq], s);
            !aiter.Done();
            aiter.Next()) {
         const fst::StdArc &arc = aiter.Value();
         DenominatorGraphTransition transition;
-        transition.transition_prob = exp(-(arc.weight.Value() - offset));
-        transition.pdf_id = arc.ilabel - 1;
+        transition.transition_prob = -(arc.weight.Value() - offset);
+        int pdf_id = arc.ilabel - 1;
```

Comment:
make this int32, and please add a comment like
// note: the FST labels were pdf-id plus one.